### PR TITLE
[guides] update build guide from autotools to cmake

### PR DESCRIPTION
**Pull request by superwhd**

### CODE REVIEW

File: site/en/guides/build/index.md [hunk 1]
```diff
@@ -84,29 +46,29 @@ locations:
...
+$ ./script/build nrf52840 UART_trans  -DOT_JAM_DETECTION=1
 ```
 
 ### Binaries
 
-The following binaries are generated in `/output/{platform}/bin` from the build process. To determine which binaries are generated, use configure option flags with the `./configure` command to generate an updated `Makefile` for building. For example, to build OpenThread and generate only the CLI binaries:
+The following binaries are generated in `/build/bin` from the build process. To determine which binaries are generated, use configure option flags with the `./configure` command to generate an updated `Makefile` for building. For example, to build OpenThread and generate only the CLI binaries:
 
 ```
 $ ./configure --enable-cli
```
Comment (superwhd) [inline site/en/guides/build/index.md:74]: I don't know if we have according switches in cmake.

File: site/en/guides/build/index.md [hunk 2]
```diff
@@ -48,32 +15,27 @@ The most common workflow is:
...
             $ ./script/bootstrap
 
-    1.  Set up the environment:
-
-            $ ./bootstrap
-
-    1.  Configure and build, using pre-defined platform examples with optional customization via common switches:
-        1.  Modify OpenThread compile-time constants in the selected platform's `/examples/platforms/{platform}/openthread-core-{platform}-config.h` file
-        1.  Build the configuration:
+    1.  Modify OpenThread compile-time constants in `/src/nrf52840/openthread-core-nrf52840-config.h` file.
```
Comment (Vyrastas) [inline site/en/guides/build/index.md:27]: ...in the

File: site/en/guides/build/index.md [hunk 3]
```diff
@@ -48,32 +15,27 @@ The most common workflow is:
...
+    1.  Build the configuration:
 
-                $ make -f examples/Makefile-{platform} <switches>
+            ./script/build nrf52840 UART_trans <platform-specific-args> <cmake-switches>
 
 1.  Flash the desired binary to the target platform. All generated binaries are
-    located in `/output/{platform}/bin`.
+    located in `/build/bin`.
 
 Specific instructions on building supported platforms with GNU Autotools can be
```
Comment (Vyrastas) [inline site/en/guides/build/index.md:35]: Should this change if we're referencing the individual platform repos now?  Will the examples/platforms directory be removed at some point?

**Review by Vyrastas — CHANGES_REQUESTED** [review]

File: site/en/guides/build/index.md (hunk 2 above)
Comment (jwhui) [inline site/en/guides/build/index.md:27]: How about we move this step to the Configuration section? For those trying OpenThread for the first time, they should just use the default configuration.

File: site/en/guides/build/index.md (hunk 3 above)
Comment (jwhui) [reply site/en/guides/build/index.md:35]: Yes. Also, we're not using autotools anymore.

File: site/en/guides/build/index.md [hunk 4]
```diff
@@ -48,32 +15,27 @@ The most common workflow is:
...
-            $ ./bootstrap
-
-    1.  Configure and build, using pre-defined platform examples with optional customization via common switches:
-        1.  Modify OpenThread compile-time constants in the selected platform's `/examples/platforms/{platform}/openthread-core-{platform}-config.h` file
-        1.  Build the configuration:
+    1.  Modify OpenThread compile-time constants in `/src/nrf52840/openthread-core-nrf52840-config.h` file.
+    1.  Build the configuration:
 
-                $ make -f examples/Makefile-{platform} <switches>
+            ./script/build nrf52840 UART_trans <platform-specific-args> <cmake-switches>
```
Comment (jwhui) [inline site/en/guides/build/index.md:30]: `UART_trans` is also specific to nrf52840
```suggestion
            ./script/build nrf52840 <platform-specific-args> <cmake-switches>
```

File: site/en/guides/build/index.md [hunk 5]
```diff
@@ -48,32 +15,27 @@ The most common workflow is:
             $ docker pull openthread/environment:latest
             $ docker run -it --rm openthread/environment bash
 
-1.  Within your chosen environment, clone the OpenThread Git repository:
+1.  Within your chosen environment, clone the platform-specific OpenThread Git repository. Taking nrf52840 as an example:
```
Comment (jwhui) [inline site/en/guides/build/index.md:18]: Maybe we should stick to cc2538 for now until  we can support the different platforms in the docs.

File: site/en/guides/build/index.md (hunk 1 above)
Comment (jwhui) [reply site/en/guides/build/index.md:74]: We have `OT_APP_CLI`, but that is slightly different in that it only controls building the app, not the CLI library itself. CMake always builds the CLI lib.

File: site/en/guides/build/index.md (hunk 4 above)
Comment (bukepo) [reply site/en/guides/build/index.md:30]: Can we make UART_trans a normal cmake option?

**Comment by superwhd** [comment]
I found that there are some links broken. For example `/guides/thread-primer/node-roles-and-types#device_types`. Shall I replace them with full URL (https://github.com/openthread/openthread/blob/main/doc/site/en/guides/thread-primer/node-roles-and-types.md)?

File: site/en/guides/build/index.md [hunk 6]
```diff
@@ -84,45 +42,44 @@ locations:
...
 
 ```
-$ make -f examples/Makefile-nrf52840 JAM_DETECTION=1
+$ ./script/build nrf52840 UART_trans  -DOT_JAM_DETECTION=1
 ```
 
 ### Binaries
 
-The following binaries are generated in `/output/{platform}/bin` from the build process. To determine which binaries are generated, use configure option flags with the `./configure` command to generate an updated `Makefile` for building. For example, to build OpenThread and generate only the CLI binaries:
+The following binaries are generated in `/build/bin` from the build process. To determine which binaries are generated, use flags with the `script/build` command. For example, to build OpenThread and generate only the FTD CLI binary:
```
Comment (librasungirl) [inline site/en/guides/build/index.md:67]: should we use the relative path instead, e.g.`./build/bin/`, `./script/build`?

File: site/en/guides/build/index.md [hunk 7]
```diff
@@ -84,45 +42,44 @@ locations:
...
 
 ```
-$ ./configure --enable-cli
-$ make
+$ ./script/build -DOT_APP_CLI=1 -DOT_FTD=1 -DOT_MTD=0 -DOT_APP_NCP=0 -DOT_APP_RCP=0 -DOT_RCP=0
 ```
 
-Binary | Description | Configure option flags
+Binary | Description | Flags
```
Comment (librasungirl) [inline site/en/guides/build/index.md:73]:
```suggestion
Binary | Description | Options
```

File: site/en/guides/build/index.md [hunk 8]
```diff
@@ -84,45 +42,44 @@ locations:
...
-If neither these flags nor a platform example are not used, applications are not
-built but OpenThread library files are still generated in `/output/{platform}/lib` for use in a project.
+`ot-cli-ftd` | Full Thread device for SoC designs | `-DOT_APP_CLI=1`<br/> `-DOT_FTD=1`
+`ot-cli-mtd` | Minimal Thread device for SoC designs | `-DOT_APP_CLI=1`<br/> `-DOT_MTD=1`
+`ot-ncp-ftd` | Full Thread device for Network Co-Processor (NCP) designs | `-DOT_APP_NCP=1`<br/> `-DOT_FTD=1`
+`ot-ncp-mtd` | Minimal Thread device for NCP designs | `-DOT_APP_NCP=1`<br/> `-DOT_MTD=1`
+`ot-rcp` | Radio Co-Processor (RCP) design | `-DOT_APP_RCP=1`<br/> `-DOT_RCP=1`
+
+By default, all above flags are enabled. If explicitly disable all flags, applications are not
+built but OpenThread library files are still generated in `/build/lib` for use in a project.
```
Comment (librasungirl) [inline site/en/guides/build/index.md:82]: ditto, use relative path `./build/lib/`?

File: site/en/guides/build/index.md [hunk 9]
```diff
@@ -84,45 +42,44 @@ locations:
...
 ---- | ----
 Compile-time constants | Listed in all the header files in [`/src/core/config`](https://github.com/openthread/openthread/tree/main/src/core/config)
-Makefile build switches | Listed in [`/examples/common-switches.mk`](https://github.com/openthread/openthread/tree/main/examples/common-switches.mk)
+Makefile build switches | Listed in [`openthread/examples/README.md`](https://github.com/openthread/openthread/blob/main/examples/README.md)
 
 > Note: Each example platform included in OpenThread specifies some, but not all, of the constants and flags that the platform supports. Modify the example platform's `/openthread-core-{platform}-config.h` file to enable or disable compile-time constants prior to building.
 
 ### Build examples
 
 Use a switch to enable functionality for an example platform. For example, to
```
Comment (librasungirl) [inline site/en/guides/build/index.md:51]:
```suggestion
Use cmake build option to enable functionality for the platform. For example, to
```

File: site/en/guides/build/index.md [hunk 10]
```diff
@@ -48,32 +15,23 @@ The most common workflow is:
...
+    1.  Build the configuration:
 
-            $ ./bootstrap
-
-    1.  Configure and build, using pre-defined platform examples with optional customization via common switches:
-        1.  Modify OpenThread compile-time constants in the selected platform's `/examples/platforms/{platform}/openthread-core-{platform}-config.h` file
-        1.  Build the configuration:
-
-                $ make -f examples/Makefile-{platform} <switches>
+            ./script/build <platform-specific-args> <cmake-switches>
```
Comment (librasungirl) [inline site/en/guides/build/index.md:29]: How about using cmake term?
```suggestion
            ./script/build <platform-specific-args> <cmake-options>
```

File: site/en/guides/build/index.md (hunk 6 above)
Comment (superwhd) [reply site/en/guides/build/index.md:67]: I'm not very sure about the convention here. I thought that the former version was using absolute paths.

File: site/en/guides/build/index.md [hunk 11]
```diff
@@ -84,45 +42,44 @@ locations:
...
-
-If neither these flags nor a platform example are not used, applications are not
-built but OpenThread library files are still generated in `/output/{platform}/lib` for use in a project.
+`ot-cli-ftd` | Full Thread device for SoC designs | `-DOT_APP_CLI=1`<br/> `-DOT_FTD=1`
+`ot-cli-mtd` | Minimal Thread device for SoC designs | `-DOT_APP_CLI=1`<br/> `-DOT_MTD=1`
+`ot-ncp-ftd` | Full Thread device for Network Co-Processor (NCP) designs | `-DOT_APP_NCP=1`<br/> `-DOT_FTD=1`
+`ot-ncp-mtd` | Minimal Thread device for NCP designs | `-DOT_APP_NCP=1`<br/> `-DOT_MTD=1`
+`ot-rcp` | Radio Co-Processor (RCP) design | `-DOT_APP_RCP=1`<br/> `-DOT_RCP=1`
+
+By default, all above flags are enabled. If explicitly disable all flags, applications are not
```
Comment (Vyrastas) [inline site/en/guides/build/index.md:81]: If you explicitly...

File: site/en/guides/build/index.md [hunk 12]
```diff
@@ -48,32 +15,23 @@ The most common workflow is:
...
-
-            $ ./bootstrap
+    1.  Build the configuration:
 
-    1.  Configure and build, using pre-defined platform examples with optional customization via common switches:
-        1.  Modify OpenThread compile-time constants in the selected platform's `/examples/platforms/{platform}/openthread-core-{platform}-config.h` file
-        1.  Build the configuration:
-
-                $ make -f examples/Makefile-{platform} <switches>
+            ./script/build <platform-specific-args> <cmake-options>
```
Comment (Vyrastas) [inline site/en/guides/build/index.md:29]: Add terminal prompt $ for this command

File: site/en/guides/build/index.md (hunk 6 above)
Comment (jwhui) [reply site/en/guides/build/index.md:67]: I think relative paths make more sense. @Vyrastas ?

File: site/en/guides/build/index.md [hunk 13]
```diff
@@ -48,32 +15,23 @@ The most common workflow is:
...
-
-            $ ./bootstrap
+    1.  Build the configuration:
 
-    1.  Configure and build, using pre-defined platform examples with optional customization via common switches:
-        1.  Modify OpenThread compile-time constants in the selected platform's `/examples/platforms/{platform}/openthread-core-{platform}-config.h` file
-        1.  Build the configuration:
-
-                $ make -f examples/Makefile-{platform} <switches>
+            $ ./script/build <platform-specific-args> <cmake-options>
```
Comment (simonlingoogle) [inline site/en/guides/build/index.md:29]: Should we use a real working command for a specific platform?

File: site/en/guides/build/index.md [hunk 14]
```diff
@@ -48,32 +15,23 @@ The most common workflow is:
             $ docker pull openthread/environment:latest
             $ docker run -it --rm openthread/environment bash
 
-1.  Within your chosen environment, clone the OpenThread Git repository:
+1.  Within your chosen environment, clone the platform-specific OpenThread Git repository. Taking CC2538 as an example:
```
Comment (simonlingoogle) [inline site/en/guides/build/index.md:18]: Switch example to nRF52840 DK?

File: site/en/guides/build/index.md [hunk 15]
```diff
@@ -48,32 +15,23 @@ The most common workflow is:
...
 
-           $ git clone https://github.com/openthread/openthread
+           $ git clone https://github.com/openthread/ot-cc2538.git --recursive
 
 1.  From the cloned repository's root directory:
-    1.  Install the GNU toolchain and other dependencies (optional):
+    1.  Install the toolchain:
```
Comment (simonlingoogle) [inline site/en/guides/build/index.md:23]: Beyond the scope of this PR, suggest to install toolchain in when building `openthread/environment` docker image.

File: site/en/guides/build/index.md [hunk 16]
```diff
@@ -84,45 +42,44 @@ locations:
 Type | Location
 ---- | ----
 Compile-time constants | Listed in all the header files in [`/src/core/config`](https://github.com/openthread/openthread/tree/main/src/core/config)
-Makefile build switches | Listed in [`/examples/common-switches.mk`](https://github.com/openthread/openthread/tree/main/examples/common-switches.mk)
-
-> Note: Each example platform included in OpenThread specifies some, but not all, of the constants and flags that the platform supports. Modify the example platform's `/openthread-core-{platform}-config.h` file to enable or disable compile-time constants prior to building.
+cmake build options | Listed in [`openthread/examples/README.md`](https://github.com/openthread/openthread/blob/main/examples/README.md)
+.
+> Note: Each platform repo specifies some, but not all, of the constants and flags that the platform supports. Modify the platform's `openthread-core-{platform}-config.h` file to enable or disable compile-time constants prior to building.
```
Comment (simonlingoogle) [inline site/en/guides/build/index.md:47]:
```suggestion
> Note: Each platform repository specifies some, but not all, of the constants and flags that the platform supports. Modify the platform's `openthread-core-{platform}-config.h` file to enable or disable compile-time constants prior to building.
```

File: site/en/guides/build/index.md [hunk 17]
```diff
@@ -84,45 +42,44 @@ locations:
...
 ### Build examples
 
-Use a switch to enable functionality for an example platform. For example, to
-build the CC2538 example with Commissioner and Joiner support enabled:
+Use cmake build option to enable functionality for the platform. For example, to
+build the binary for the CC2538 platform with Commissioner and Joiner support enabled:
 
 ```
-$ make -f examples/Makefile-cc2538 COMMISSIONER=1 JOINER=1
+$ ./script/build -DOT_COMMISSIONER=1 -DOT_JOINER=1
```
Comment (simonlingoogle) [inline site/en/guides/build/index.md:55]: Reduce cc2538 example to reduce maintenance overhead?

@jwhui @librasungirl

File: site/en/guides/build/index.md (hunk 13 above)
Comment (librasungirl) [reply site/en/guides/build/index.md:29]: Here seems a general guide, there is build example below

File: site/en/guides/build/index.md (hunk 6 above)
Comment (librasungirl) [reply site/en/guides/build/index.md:67]: > I'm not very sure about the convention here. I thought that the former version was using absolute paths.

Here are the build output and script directory, both are relative to the source repo directory, e.g,  
you may find the full path for build output would be `<path-to-src repo>/build`

File: site/en/guides/build/index.md [hunk 18]
```diff
@@ -84,45 +42,44 @@ locations:
...
```
Comment (librasungirl) [inline site/en/guides/build/index.md:55]:
```suggestion
$ ./script/build -DOT_COMMISSIONER=ON -DOT_JOINER=ON
```

File: site/en/guides/build/index.md [hunk 19]
```diff
@@ -84,45 +42,44 @@ locations:
...
 ```
 
-Or, to build the nRF52840 example with the [Jam Detection
-feature](/guides/build/features/jam-detection) enabled:
+Or, to build the nRF52840 platform with the [Jam Detection
+feature](/guides/build/features/jam-detection) enabled in its repo:
 
 ```
-$ make -f examples/Makefile-nrf52840 JAM_DETECTION=1
+$ ./script/build nrf52840 UART_trans  -DOT_JAM_DETECTION=1
```
Comment (librasungirl) [inline site/en/guides/build/index.md:62]:
```suggestion
$ ./script/build nrf52840 UART_trans  -DOT_JAM_DETECTION=ON
```

File: site/en/guides/build/index.md [hunk 20]
```diff
@@ -84,45 +42,44 @@ locations:
...
 
 ### Binaries
 
-The following binaries are generated in `/output/{platform}/bin` from the build process. To determine which binaries are generated, use configure option flags with the `./configure` command to generate an updated `Makefile` for building. For example, to build OpenThread and generate only the CLI binaries:
+The following binaries are generated in `./build/bin` from the build process. To determine which binaries are generated, use flags with the `./script/build` command. For example, to build OpenThread and generate only the FTD CLI binary:
 
 ```
-$ ./configure --enable-cli
-$ make
+$ ./script/build -DOT_APP_CLI=1 -DOT_FTD=1 -DOT_MTD=0 -DOT_APP_NCP=0 -DOT_APP_RCP=0 -DOT_RCP=0
```
Comment (librasungirl) [inline site/en/guides/build/index.md:70]:
```suggestion
$ ./script/build -DOT_APP_CLI=ON -DOT_FTD=ON -DOT_MTD=OFF -DOT_APP_NCP=OFF -DOT_APP_RCP=OFF -DOT_RCP=OFF
```

File: site/en/guides/build/index.md [hunk 21]
```diff
@@ -84,45 +42,44 @@ locations:
...
 ---- | ---- | ----
-`ot-cli-ftd` | Full Thread device for SoC designs | `--enable-cli`<br/> `--enable-ftd`
-`ot-cli-mtd` | Minimal Thread device for SoC designs | `--enable-cli`<br/> `--enable-mtd`
-`ot-ncp-ftd` | Full Thread device for Network Co-Processor (NCP) designs | `--enable-ncp`<br/> `--enable-ftd`
-`ot-ncp-mtd` | Minimal Thread device for NCP designs | `--enable-ncp`<br/> `--enable-mtd`
-`ot-rcp` | Radio Co-Processor (RCP) design | `--enable-ncp`<br/> `--enable-radio-only`
-
-If neither these flags nor a platform example are not used, applications are not
-built but OpenThread library files are still generated in `/output/{platform}/lib` for use in a project.
+`ot-cli-ftd` | Full Thread device for SoC designs | `-DOT_APP_CLI=1`<br/> `-DOT_FTD=1`
```
Comment (librasungirl) [inline site/en/guides/build/index.md:75]:
```suggestion
`ot-cli-ftd` | Full Thread device for SoC designs | `-DOT_APP_CLI=ON`<br/> `-DOT_FTD=ON`
```

File: site/en/guides/build/index.md [hunk 22]
```diff
@@ -84,45 +42,44 @@ locations:
...
+`ot-cli-mtd` | Minimal Thread device for SoC designs | `-DOT_APP_CLI=1`<br/> `-DOT_MTD=1`
```
Comment (librasungirl) [inline site/en/guides/build/index.md:76]:
```suggestion
`ot-cli-mtd` | Minimal Thread device for SoC designs | `-DOT_APP_CLI=ON`<br/> `-DOT_MTD=ON`
```

File: site/en/guides/build/index.md [hunk 23]
```diff
@@ -84,45 +42,44 @@ locations:
...
+`ot-ncp-ftd` | Full Thread device for Network Co-Processor (NCP) designs | `-DOT_APP_NCP=1`<br/> `-DOT_FTD=1`
```
Comment (librasungirl) [inline site/en/guides/build/index.md:77]:
```suggestion
`ot-ncp-ftd` | Full Thread device for Network Co-Processor (NCP) designs | `-DOT_APP_NCP=ON`<br/> `-DOT_FTD=ON`
```

File: site/en/guides/build/index.md [hunk 24]
```diff
@@ -84,45 +42,44 @@ locations:
...
+`ot-ncp-mtd` | Minimal Thread device for NCP designs | `-DOT_APP_NCP=1`<br/> `-DOT_MTD=1`
```
Comment (librasungirl) [inline site/en/guides/build/index.md:78]:
```suggestion
`ot-ncp-mtd` | Minimal Thread device for NCP designs | `-DOT_APP_NCP=ON`<br/> `-DOT_MTD=ON`
```

File: site/en/guides/build/index.md [hunk 25]
```diff
@@ -84,45 +42,44 @@ locations:
...
+`ot-rcp` | Radio Co-Processor (RCP) design | `-DOT_APP_RCP=1`<br/> `-DOT_RCP=1`
```
Comment (librasungirl) [inline site/en/guides/build/index.md:79]:
```suggestion
`ot-rcp` | Radio Co-Processor (RCP) design | `-DOT_APP_RCP=ON`<br/> `-DOT_RCP=ON`
```

**Review by bukepo — APPROVED** [review]

**Review by librasungirl — APPROVED** [review]

File: site/en/guides/build/index.md [hunk 26]
```diff
@@ -84,45 +42,44 @@ locations:
 Type | Location
 ---- | ----
 Compile-time constants | Listed in all the header files in [`/src/core/config`](https://github.com/openthread/openthread/tree/main/src/core/config)
-Makefile build switches | Listed in [`/examples/common-switches.mk`](https://github.com/openthread/openthread/tree/main/examples/common-switches.mk)
-
-> Note: Each example platform included in OpenThread specifies some, but not all, of the constants and flags that the platform supports. Modify the example platform's `/openthread-core-{platform}-config.h` file to enable or disable compile-time constants prior to building.
+cmake build options | Listed in [`openthread/examples/README.md`](https://github.com/openthread/openthread/blob/main/examples/README.md)
+.
```
Comment (Vyrastas) [inline site/en/guides/build/index.md:46]: Remove this period, this should be a blank line

**Review by Vyrastas — CHANGES_REQUESTED** [review]

File: site/en/guides/build/index.md [hunk 27]
```diff
@@ -48,32 +15,23 @@ The most common workflow is:
...
+    1.  Build the configuration:
 
-            $ ./bootstrap
-
-    1.  Configure and build, using pre-defined platform examples with optional customization via common switches:
-        1.  Modify OpenThread compile-time constants in the selected platform's `/examples/platforms/{platform}/openthread-core-{platform}-config.h` file
-        1.  Build the configuration:
-
-                $ make -f examples/Makefile-{platform} <switches>
+            $ ./script/build <platform-specific-args> <cmake-options>
```
Comment (Vyrastas) [inline site/en/guides/build/index.md:29]:
```suggestion
            $ ./script/build {platform-specific-args} {cmake-options}
```

**Comment by Vyrastas** [comment]
Note: I updated the replaceable variables in an example command to use { } instead of < >

This is needed for the DevSite import

**Review by Vyrastas — APPROVED** [review]

File: site/en/guides/build/index.md [hunk 28]
```diff
@@ -84,45 +42,44 @@ locations:
...
 ```
 
-Or, to build the nRF52840 example with the [Jam Detection
-feature](/guides/build/features/jam-detection) enabled:
+Or, to build the nRF52840 platform with the [Jam Detection
+feature](/guides/build/features/jam-detection) enabled in its repo:
 
 ```
-$ make -f examples/Makefile-nrf52840 JAM_DETECTION=1
+$ ./script/build nrf52840 UART_trans  -DOT_JAM_DETECTION=ON
```
Comment (jwhui) [inline site/en/guides/build/index.md:62]:
```suggestion
$ ./script/build nrf52840 UART_trans -DOT_JAM_DETECTION=ON
```

File: site/en/guides/build/index.md [hunk 29]
```diff
@@ -84,45 +42,44 @@ locations:
...
+cmake build options | Listed in [`openthread/examples/README.md`](https://github.com/openthread/openthread/blob/main/examples/README.md)
 
-> Note: Each example platform included in OpenThread specifies some, but not all, of the constants and flags that the platform supports. Modify the example platform's `/openthread-core-{platform}-config.h` file to enable or disable compile-time constants prior to building.
+> Note: Each platform repository specifies some, but not all, of the constants and flags that the platform supports. Modify the platform's `openthread-core-{platform}-config.h` file to enable or disable compile-time constants prior to building.
 
 ### Build examples
 
-Use a switch to enable functionality for an example platform. For example, to
-build the CC2538 example with Commissioner and Joiner support enabled:
+Use cmake build option to enable functionality for the platform. For example, to
```
Comment (jwhui) [inline site/en/guides/build/index.md:51]:
```suggestion
Use cmake build options to enable functionality for the platform. For example, to
```

**Pull request merged**